subprojects/core/src/main/java/org/gradle/api/internal/artifacts/dsl/dependencies/DefaultDependencyCollector.java

```diff
@@ -18,44 +18,63 @@
 
 import org.gradle.api.Action;
 import org.gradle.api.InvalidUserCodeException;
-import org.gradle.api.artifacts.Configuration;
 import org.gradle.api.artifacts.Dependency;
 import org.gradle.api.artifacts.ExternalModuleDependency;
 import org.gradle.api.artifacts.FileCollectionDependency;
 import org.gradle.api.artifacts.MinimalExternalModuleDependency;
-import org.gradle.api.artifacts.dsl.DependencyAdder;
+import org.gradle.api.artifacts.dsl.DependencyCollector;
 import org.gradle.api.file.FileCollection;
+import org.gradle.api.internal.artifacts.dependencies.AbstractModuleDependency;
+import org.gradle.api.internal.provider.PropertyInternal;
 import org.gradle.api.provider.Provider;
 import org.gradle.api.provider.ProviderConvertible;
+import org.gradle.api.provider.SetProperty;
+import org.gradle.api.tasks.Nested;
 
 import javax.annotation.Nullable;
 import javax.inject.Inject;
 import java.util.ArrayList;
 import java.util.Collection;
 import java.util.List;
 
-public class DefaultDependencyAdder implements DependencyAdder {
+public abstract class DefaultDependencyCollector implements DependencyCollector {
     private final DependencyFactoryInternal dependencyFactory;
-    private final Configuration configuration;
 
     @Inject
-    public DefaultDependencyAdder(DependencyFactoryInternal dependencyFactory, Configuration configuration) {
+    public DefaultDependencyCollector(DependencyFactoryInternal dependencyFactory) {
         this.dependencyFactory = dependencyFactory;
-        this.configuration = configuration;
+        this.getDependencies().finalizeValueOnRead();
     }
 
+    @Nested
+    public abstract SetProperty<Dependency> getDependencies();
```

Comment:
This is a very welcome change, @jvandort, thank you!
🎉 Now we can declare our own configurations coupled with `JvmTestSuite`s in addition to the built-in 4.

Example where I migrated from a hacky workaround to public (incubating) APIs:
https://github.com/TWiStErRob/net.twisterrob.cinema/pull/633/files

This enables:
```
testing.suites.... { dependencies { kapt(project.libs.dagger.apt) } }
```

Hopefully this will and in KGP itself one day.